src/features/loading/status-announcer.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ModelViewerElementBase from '../../model-viewer-base.js';
+import {debounce, getFirstMapKey} from '../../utils.js';
+
+export const INITIAL_STATUS_ANNOUNCEMENT =
+    'This page includes one or more 3D models that are loading';
+export const FINISHED_LOADING_ANNOUNCEMENT =
+    'All 3D models in the page have loaded';
+export const UPDATE_STATUS_DEBOUNCE_MS = 100;
+
+
+const $modelViewerStatusInstance = Symbol('modelViewerStatusInstance');
+const $updateStatus = Symbol('updateStatus');
+
+interface InstanceLoadingStatus {
+  instanceWasUnregistered: () => void;
+  loadAttemptCompletes: Promise<any>;
+}
+
+/**
+ * The LoadingStatusAnnouncer manages announcements of loading status across
+ * all <model-viewer> elements in the document at any given time. As new
+ * <model-viewer> elements are connected to the document, they are registered
+ * with a LoadingStatusAnnouncer singleton. As they are disconnected, the are
+ * also unregistered. Announcements are made to indicate the following
+ * conditions:
+ *
+ *  1. There are <model-viewer> elements that have yet to finish loading
+ *  2. All <model-viewer> elements in the page have finished attempting to load
+ */
+export class LoadingStatusAnnouncer {
+  /**
+   * The "status" instance is the <model-viewer> instance currently designated
```

Comment:
bikeshed: I initially assumed that status meant something else, although this comment cleared it up for me.

Would "announcer" be clearer?

Reply:
Ooh, I see - https://www.w3.org/TR/wai-aria-1.1/#status (linked below) gives a bit more context. I'll leave this comment here but I feel even less strongly about this bikeshed.

An idea might be "statusAnnouncer" or something, but that's cumbersome.

Reply:
Yeah, I was trying to stick to the ARIA terminology. I'll ponder changing it.